File: apzkr-pzpi-21-11-milennyi-ivan/Task3-WebClient/src/viewModels/SheepVM.ts
import { timestampToDate } from "../utils/helpers";

// Клас для відображення інформації про вівцю
export class SheepVM {
    id: number; // Ідентифікатор вівці
    breed: string; // Порода вівці
    sex: boolean; // Стать вівці (true - самець, false - самка)
    birthDate: number; // Дата народження вівці
    lastFeedingDate: string | null; // Дата останнього годування
    lastShearingDate: string | null; // Дата останньої стрижки

    constructor(id: number, breed: string, sex: boolean, birthDate: number, lastFeedingTimestamp: number | null, lastShearingTimestamp: number | null) {
        this.id = id;
        this.breed = breed;
        this.sex = sex;
        this.birthDate = birthDate;
        this.lastFeedingDate = (lastFeedingTimestamp != null) ? timestampToDate(lastFeedingTimestamp) : null;
        this.lastShearingDate = (lastShearingTimestamp != null) ? timestampToDate(lastShearingTimestamp) : null;
    }
}

File: apzkr-pzpi-21-11-milennyi-ivan/Task1-Server/src/view_models/sheep_vm.rs
use serde::Serialize;
use sqlx::FromRow;
use utoipa::ToSchema;

// Структура, що представляє представлення вівці (ViewModel)
#[derive(Debug, Serialize, FromRow, ToSchema)]
pub(crate) struct SheepVM {
    id: u64, // Ідентифікатор вівці
    breed: String, // Порода вівці
    sex: bool, // Стать вівці
    birth_date: u64, // Дата народження (у вигляді таймстемпу)
    last_feeding_timestamp: Option<u64>, // Час останнього годування (може бути відсутнім)
    last_shearing_timestamp: Option<u64>, // Час останньої стрижки (може бути відсутнім)
}
